### ISSUE TITLE
Cannot do exclusion on field _id in inclusion projection v2.6

### ISSUE THREAD

**Issue by j0nd0n7** [meteor/meteor#11892]
Excluding _id field `{_id: 0, other_field: 1}` for the returned projection of a find throws the error in version 2.6:

`Cannot do exclusion on field _id in inclusion projection v2.6`

It works up to 2.5.6.

- OS: MacOs 10.14.6
- MongoDB: MongoDB 5.0.5 Community
- The expected behavior.
Returns the results without _id field and with the other fields

**Comment by StorytellerCZ** [comment]
You can either specify fields to be included or excluded, not mix it from what I remember. I'm surprised that it worked up till now. Maybe it did not, but Meteor just automatically excluded any mention of `_id` as it is needed for things to work properly and then handled it.
This is a comment at the error:
```
// this error: "Cannot do exclusion on field _id in inclusion projection"
// happens on MongoDB CLI but doesn't happen in the Node.js Driver for MongoDB 5.0+
```
So this is not a bug, but at the very least an issue with upgrade of MongoDB.

**Comment by radekmie** [comment]
> You can either specify fields to be included or excluded, not mix it from what I remember.

That's true, but it's different for `_id` ([docs](https://docs.mongodb.com/manual/tutorial/project-fields-from-query-results/#suppress-_id-field)).

**Comment by renanccastro** [comment]
hi @j0nd0n7 , thanks for reporting.
Yes, that was introduced in the new mongo package version by us, on purpose.

The justification for this behavior was that on "mongosh" or "mongo" shell client, find excluding _id is not allowed in oppose to what states in the documentation that @radekmie linked.

We decided at the moment to throw an error to mimic the behavior of the mongo shell, as there is already validation for observables without _id's inside meteor, which is not allowed.

I'm reverting this change to be compatible with previous versions, as the mongo driver allows it, and also older meteor versions support this behavior.

Sorry for the trouble, this will be available in the next patch version of the mongo package, which I will warn here when it's out.

----
But out of curiosity, what is your use case for omitting the _id in the query? I find it quite hard to think in a scenario where this is useful, as the MongoDB server always fetches the _id and it's the driver that omits it.

**Comment by radekmie** [comment]
We do it quite often:
1. As a `distinct` alternative when we know that values are unique and we'd rather stream them instead.
2. To aggregate data for batch checks, like "give me a list of allowed date ranges". In such cases, we don't care about IDs. Similarly, something like "give me my token" - an `_id`-only query only for the `token` field (`Users.findOne({ _id }, { fields: { _id: 0, token: 1 } })?.token`).
3. Fairly often with analytical aggregations, where the ID has no business meaning and we'd rather fetch fewer data.

**Comment by renanccastro** [comment]
I've Published mongo@1.14.3 with this fix.

Thanks!
